File: src/screens/PrivacyControlsScreen.js
import { Ionicons } from '@expo/vector-icons';
import { doc, getDoc } from 'firebase/firestore';
import { useEffect, useRef, useState } from 'react';
import { Alert, Image, ScrollView, StatusBar, StyleSheet, Switch, Text, TouchableOpacity, View } from 'react-native';
import { SafeAreaView } from 'react-native-safe-area-context';
import { db } from '../config/firebase';
import { useTheme } from '../context/ThemeContext';
import { useUser } from '../context/UserContext';

export default function PrivacyControlsScreen({ navigation }) {
  const { theme } = useTheme();
  const { userData, updatePrivacySettings, unblockUser, unmuteUser } = useUser(); // ✅ Added unmuteUser

  const [showVisibilityPicker, setShowVisibilityPicker] = useState(false);
  const [showFollowPicker, setShowFollowPicker] = useState(false);
  const [showCommentPicker, setShowCommentPicker] = useState(false);
  const [showClubsPicker, setShowClubsPicker] = useState(false);

  // Fetch display names + avatars for blocked/muted user IDs
  const [userCache, setUserCache] = useState({});
  const fetchedRef = useRef(new Set());
  useEffect(() => {
    const allIds = [...(userData.blocked || []), ...(userData.mutedUsers || [])];
    const toFetch = allIds.filter(id => !fetchedRef.current.has(id));
    if (toFetch.length === 0) return;
    toFetch.forEach(id => fetchedRef.current.add(id));
    Promise.all(toFetch.map(async id => {
      try {
        const snap = await getDoc(doc(db, 'users', id));
        if (snap.exists()) return [id, { name: snap.data().name || 'Unknown', avatar: snap.data().avatar || null }];
      } catch { }
      return [id, { name: 'Unknown', avatar: null }];
    })).then(results => {
      const updates = {};
      results.forEach(([id, data]) => { updates[id] = data; });
      setUserCache(prev => ({ ...prev, ...updates }));
    });
  }, [userData.blocked, userData.mutedUsers]);

  // Get current settings with defaults
  const settings = userData.privacySettings || {
    profileVisibility: 'public',
    showActivityOnFeed: true,
    showLocationOnMap: true,
    showStatsToOthers: true,
    whoCanFollow: 'everyone',
    whoCanComment: 'everyone',
    whoCanSeeClubs: 'everyone'
  };

  const blockedUsers = userData.blocked || [];
  const mutedUsers = userData.mutedUsers || []; // ✅ Extract muted users

  // Helper to update specific setting
  const toggleSetting = (key, value) => {
    updatePrivacySettings({ [key]: value });
  };

  const handleUnblock = (userId) => {
    Alert.alert(
      'Unblock User',
      'Are you sure you want to unblock this user?',
      [
        { text: 'Cancel', style: 'cancel' },
        { text: 'Unblock', style: 'destructive', onPress: () => unblockUser(userId) }
      ]
    );
  };

  const handleUnmute = (userId) => {
    Alert.alert(
      'Unmute User',
      'Are you sure you want to unmute this user? Their posts will reappear on your feed.',
      [
        { text: 'Cancel', style: 'cancel' },
        { text: 'Unmute', style: 'default', onPress: () => unmuteUser(userId) }
      ]
    );
  };

  const OptionRow = ({ label, desc, value, onValueChange }) => (
    <View style={[styles.row, { borderBottomColor: theme.colors.border }]}>
      <View style={{ flex: 1, paddingRight: 10 }}>
        <Text style={[styles.label, { color: theme.colors.text }]}>{label}</Text>
        <Text style={[styles.desc, { color: theme.colors.subText }]}>{desc}</Text>
      </View>
      <Switch
        value={value}
        onValueChange={onValueChange}
        trackColor={{ true: theme.colors.accent, false: '#333' }}
        thumbColor="#FFF"
      />
    </View>
  );

  const SelectRow = ({ label, desc, value, options, onSelect, showPicker, setShowPicker }) => (
    <>
      <TouchableOpacity
        style={[styles.row, { borderBottomColor: theme.colors.border }]}
        onPress={() => setShowPicker(!showPicker)}
      >
        <View style={{ flex: 1, paddingRight: 10 }}>
          <Text style={[styles.label, { color: theme.colors.text }]}>{label}</Text>
          <Text style={[styles.desc, { color: theme.colors.subText }]}>{desc}</Text>
        </View>
        <View style={styles.selectValue}>
          <Text style={[styles.valueText, { color: theme.colors.accent }]}>
            {value === 'public' ? 'Public' : value === 'friends' ? 'Friends Only' : value === 'private' ? 'Private' : value === 'everyone' ? 'Everyone' : value === 'nobody' ? 'Nobody' : value.charAt(0).toUpperCase() + value.slice(1)}
          </Text>
          <Ionicons name={showPicker ? 'chevron-up' : 'chevron-down'} size={20} color="#666" />
        </View>
      </TouchableOpacity>

      {showPicker && (
        <View style={[styles.pickerContainer, { backgroundColor: theme.colors.card }]}>
          {options.map((option) => (
            <TouchableOpacity
              key={option.value}
              style={styles.pickerOption}
              onPress={() => {
                onSelect(option.value);
                setShowPicker(false);
              }}
            >
              <Text style={[styles.pickerText, { color: theme.colors.text }]}>{option.label}</Text>
              {value === option.value && <Ionicons name="checkmark" size={20} color={theme.colors.accent} />}
            </TouchableOpacity>
          ))}
        </View>
      )}
    </>
  );

  return (
    <View style={[styles.container, { backgroundColor: theme.colors.background }]}>
      <StatusBar barStyle={theme.dark ? "light-content" : "dark-content"} />
      <SafeAreaView style={{ flex: 1 }}>
        <View style={[styles.header, { borderBottomColor: theme.colors.border }]}>
          <TouchableOpacity onPress={() => navigation.goBack()}>
            <Ionicons name="arrow-back" size={24} color={theme.colors.text} />
          </TouchableOpacity>
          <Text style={[styles.title, { color: theme.colors.text }]}>Privacy Controls</Text>
          <View style={{ width: 24 }} />
        </View>

        <ScrollView contentContainerStyle={styles.content}>
          {/* PROFILE VISIBILITY */}
          <Text style={[styles.sectionTitle, { color: theme.colors.accent }]}>PROFILE VISIBILITY</Text>
          <View style={[styles.card, { backgroundColor: theme.colors.card }]}>
            <SelectRow
              label="Who can see my profile"
              desc="Control who can view your profile and achievements."
              value={settings.profileVisibility || 'public'}
              options={[
                { label: 'Public', value: 'public' },
                { label: 'Friends Only', value: 'friends' },
                { label: 'Private', value: 'private' }
              ]}
              onSelect={(val) => toggleSetting('profileVisibility', val)}
              showPicker={showVisibilityPicker}
              setShowPicker={setShowVisibilityPicker}
            />
          </View>

          {/* ACTIVITY SETTINGS */}
          <Text style={[styles.sectionTitle, { color: theme.colors.accent }]}>ACTIVITY SETTINGS</Text>
          <View style={[styles.card, { backgroundColor: theme.colors.card }]}>
            <OptionRow
              label="Show Activity on Feed"
              desc="Allow your runs to appear on the community feed."
              value={settings.showActivityOnFeed !== false}
              onValueChange={(val) => toggleSetting('showActivityOnFeed', val)}
            />
            <OptionRow
              label="Show Location on Map"
              desc="Display your GPS route on public feeds."
              value={settings.showLocationOnMap !== false}
              onValueChange={(val) => toggleSetting('showLocationOnMap', val)}
            />
            <OptionRow
              label="Show Stats to Others"
              desc="Allow others to see your detailed statistics."
              value={settings.showStatsToOthers !== false}
              onValueChange={(val) => toggleSetting('showStatsToOthers', val)}
            />
          </View>

          {/* SOCIAL PERMISSIONS */}
          <Text style={[styles.sectionTitle, { color: theme.colors.accent }]}>SOCIAL PERMISSIONS</Text>
          <View style={[styles.card, { backgroundColor: theme.colors.card }]}>
            <SelectRow
              label="Who can follow me"
              desc="Control who can follow your profile."
              value={settings.whoCanFollow || 'everyone'}
              options={[
                { label: 'Everyone', value: 'everyone' },
                { label: 'Friends Only', value: 'friends' },
                { label: 'Nobody', value: 'nobody' }
              ]}
              onSelect={(val) => toggleSetting('whoCanFollow', val)}
              showPicker={showFollowPicker}
              setShowPicker={setShowFollowPicker}
            />
            <SelectRow
              label="Who can comment"
              desc="Control who can comment on your runs."
              value={settings.whoCanComment || 'everyone'}
              options={[
                { label: 'Everyone', value: 'everyone' },
                { label: 'Friends Only', value: 'friends' },
                { label: 'Nobody', value: 'nobody' }
              ]}
              onSelect={(val) => toggleSetting('whoCanComment', val)}
              showPicker={showCommentPicker}
              setShowPicker={setShowCommentPicker}
            />
            <SelectRow
              label="Who can see my clubs"
              desc="Control who can see which clubs you've joined."
              value={settings.whoCanSeeClubs || 'everyone'}
              options={[
                { label: 'Everyone', value: 'everyone' },
                { label: 'Friends Only', value: 'friends' }
              ]}
              onSelect={(val) => toggleSetting('whoCanSeeClubs', val)}
              showPicker={showClubsPicker}
              setShowPicker={setShowClubsPicker}
            />
          </View>

          {/* BLOCKED USERS */}
          <Text style={[styles.sectionTitle, { color: theme.colors.accent }]}>BLOCKED USERS</Text>
          <View style={[styles.card, { backgroundColor: theme.colors.card }]}>
            {blockedUsers.length === 0 ? (
              <View style={styles.emptyState}>
                <Ionicons name="shield-checkmark-outline" size={40} color="#666" />
                <Text style={[styles.emptyText, { color: theme.colors.subText }]}>No blocked users</Text>
              </View>
            ) : (
              blockedUsers.map((userId, index) => {
                const info = userCache[userId];
                return (
                  <View
                    key={userId}
                    style={[
                      styles.blockedUserRow,
                      { borderBottomColor: theme.colors.border },
                      index === blockedUsers.length - 1 && { borderBottomWidth: 0 }
                    ]}
                  >
                    <View style={styles.blockedUserInfo}>
                      {info?.avatar ? (
                        <Image source={{ uri: info.avatar }} style={styles.blockedAvatar} />
                      ) : (
                        <View style={[styles.blockedAvatar, { justifyContent: 'center', alignItems: 'center' }]}>
                          <Ionicons name="person" size={20} color="#666" />
                        </View>
                      )}
                      <Text style={[styles.blockedUserName, { color: theme.colors.text }]}>
                        {info ? info.name : `User ${userId.slice(0, 8)}`}
                      </Text>
                    </View>
                    <TouchableOpacity style={styles.unblockBtn} onPress={() => handleUnblock(userId)}>
                      <Text style={styles.unblockText}>Unblock</Text>
                    </TouchableOpacity>
                  </View>
                );
              })
            )}
          </View>

          <Text style={[styles.helperText, { color: theme.colors.subText }]}>
            Blocked users cannot see your profile, follow you, or interact with your content.
          </Text>

          {/* MUTED USERS (PHASE 19) */}
          <Text style={[styles.sectionTitle, { color: theme.colors.accent, marginTop: 30 }]}>MUTED USERS</Text>
          <View style={[styles.card, { backgroundColor: theme.colors.card }]}>
            {mutedUsers.length === 0 ? (
              <View style={styles.emptyState}>
                <Ionicons name="volume-medium-outline" size={40} color="#666" />
                <Text style={[styles.emptyText, { color: theme.colors.subText }]}>No muted users</Text>
              </View>
            ) : (
              mutedUsers.map((userId, index) => {
                const info = userCache[userId];
                return (
                  <View
                    key={userId}
                    style={[
                      styles.blockedUserRow,
                      { borderBottomColor: theme.colors.border },
                      index === mutedUsers.length - 1 && { borderBottomWidth: 0 }
                    ]}
                  >
                    <View style={styles.blockedUserInfo}>
                      {info?.avatar ? (
                        <Image source={{ uri: info.avatar }} style={styles.blockedAvatar} />
                      ) : (
                        <View style={[styles.blockedAvatar, { justifyContent: 'center', alignItems: 'center' }]}>
                          <Ionicons name="person" size={20} color="#666" />
                        </View>
                      )}
                      <Text style={[styles.blockedUserName, { color: theme.colors.text }]}>
                        {info ? info.name : `User ${userId.slice(0, 8)}`}
                      </Text>
                    </View>
                    <TouchableOpacity
                      style={[styles.unblockBtn, { backgroundColor: theme.colors.border }]}
                      onPress={() => handleUnmute(userId)}
                    >
                      <Text style={[styles.unblockText, { color: theme.colors.text }]}>Unmute</Text>
                    </TouchableOpacity>
                  </View>
                );
              })
            )}
          </View>

          <Text style={[styles.helperText, { color: theme.colors.subText }]}>
            Muted users will not appear on your feed, but they can still see your profile and interact with you.
          </Text>
        </ScrollView>
      </SafeAreaView>
    </View>
  );
}

const styles = StyleSheet.create({
  container: { flex: 1 },
  header: { flexDirection: 'row', justifyContent: 'space-between', alignItems: 'center', padding: 20, borderBottomWidth: 1 },
  title: { fontSize: 18, fontFamily: 'Poppins_700Bold' },
  content: { padding: 20, paddingBottom: 40 },
  sectionTitle: { fontSize: 12, fontFamily: 'Poppins_700Bold', marginBottom: 10, marginTop: 10, letterSpacing: 1 },
  card: { borderRadius: 12, overflow: 'hidden', marginBottom: 20 },
  row: { flexDirection: 'row', justifyContent: 'space-between', alignItems: 'center', padding: 16, borderBottomWidth: 1 },
  label: { fontSize: 16, fontFamily: 'Poppins_600SemiBold' },
  desc: { fontSize: 12, fontFamily: 'Poppins_400Regular', marginTop: 4, lineHeight: 16 },
  selectValue: { flexDirection: 'row', alignItems: 'center', gap: 8 },
  valueText: { fontSize: 14, fontFamily: 'Poppins_600SemiBold' },
  pickerContainer: { paddingVertical: 8 },
  pickerOption: { flexDirection: 'row', justifyContent: 'space-between', alignItems: 'center', paddingVertical: 12, paddingHorizontal: 16 },
  pickerText: { fontSize: 15, fontFamily: 'Poppins_500Medium' },
  emptyState: { alignItems: 'center', paddingVertical: 40 },
  emptyText: { fontSize: 14, fontFamily: 'Poppins_400Regular', marginTop: 10 },
  blockedUserRow: { flexDirection: 'row', justifyContent: 'space-between', alignItems: 'center', padding: 16, borderBottomWidth: 1 },
  blockedUserInfo: { flexDirection: 'row', alignItems: 'center' },
  blockedAvatar: { width: 40, height: 40, borderRadius: 20, backgroundColor: '#333', marginRight: 12, overflow: 'hidden' },
  blockedUserName: { fontSize: 15, fontFamily: 'Poppins_600SemiBold' },
  unblockBtn: { backgroundColor: '#FF3B30', paddingVertical: 6, paddingHorizontal: 16, borderRadius: 6 },
  unblockText: { color: '#FFF', fontSize: 13, fontFamily: 'Poppins_600SemiBold' },
  helperText: { fontSize: 12, fontFamily: 'Poppins_400Regular', marginTop: 5, lineHeight: 18, paddingHorizontal: 5 }
});

File: android/app/src/main/java/com/ruvo/app/features/settings/PrivacyControlsScreen.kt
package com.ruvo.app.features.settings

import androidx.compose.foundation.*
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.*
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.*
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.layout.ContentScale
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.hilt.navigation.compose.hiltViewModel
import androidx.lifecycle.ViewModel
import androidx.lifecycle.compose.collectAsStateWithLifecycle
import androidx.lifecycle.viewModelScope
import coil.compose.AsyncImage
import com.google.firebase.auth.FirebaseAuth
import com.google.firebase.firestore.FieldValue
import com.google.firebase.firestore.FirebaseFirestore
import com.ruvo.app.designsystem.theme.RuvoColors
import com.ruvo.app.designsystem.theme.RuvoRadius
import com.ruvo.app.designsystem.theme.RuvoSpacing
import dagger.hilt.android.lifecycle.HiltViewModel
import kotlinx.coroutines.async
import kotlinx.coroutines.awaitAll
import kotlinx.coroutines.coroutineScope
import kotlinx.coroutines.flow.*
import kotlinx.coroutines.launch
import kotlinx.coroutines.tasks.await
import javax.inject.Inject

data class PrivacySettings(
    val profileVisibility: String = "public",
    val showActivityOnFeed: Boolean = true,
    val showLocationOnMap: Boolean = true,
    val showStatsToOthers: Boolean = true,
    val whoCanFollow: String = "everyone",
    val whoCanComment: String = "everyone",
    val whoCanSeeClubs: String = "everyone",
)

data class BlockedUserInfo(val id: String, val name: String, val avatarUrl: String?)

data class PrivacyUiState(
    val settings: PrivacySettings = PrivacySettings(),
    val blockedUsers: List<BlockedUserInfo> = emptyList(),
    val mutedUsers: List<BlockedUserInfo> = emptyList(),
)

@HiltViewModel
class PrivacyViewModel @Inject constructor(
    private val firestore: FirebaseFirestore,
    private val auth: FirebaseAuth,
) : ViewModel() {

    private val _uiState = MutableStateFlow(PrivacyUiState())
    val uiState: StateFlow<PrivacyUiState> = _uiState.asStateFlow()

    init { load() }

    private fun load() {
        viewModelScope.launch {
            val uid = auth.currentUser?.uid ?: return@launch
            try {
                val doc = firestore.collection("users").document(uid).get().await()
                val data = doc.data ?: return@launch
                val p = data["privacySettings"] as? Map<*, *>
                val cur = PrivacySettings()
                _uiState.update {
                    it.copy(
                        settings = cur.copy(
                            profileVisibility = p?.get("profileVisibility") as? String ?: cur.profileVisibility,
                            showActivityOnFeed = p?.get("showActivityOnFeed") as? Boolean ?: cur.showActivityOnFeed,
                            showLocationOnMap = p?.get("showLocationOnMap") as? Boolean ?: cur.showLocationOnMap,
                            showStatsToOthers = p?.get("showStatsToOthers") as? Boolean ?: cur.showStatsToOthers,
                            whoCanFollow = p?.get("whoCanFollow") as? String ?: cur.whoCanFollow,
                            whoCanComment = p?.get("whoCanComment") as? String ?: cur.whoCanComment,
                            whoCanSeeClubs = p?.get("whoCanSeeClubs") as? String ?: cur.whoCanSeeClubs,
                        )
                    )
                }
                @Suppress("UNCHECKED_CAST")
                val blockedIds = (data["blocked"] as? List<String>) ?: emptyList()
                @Suppress("UNCHECKED_CAST")
                val mutedIds = (data["mutedUsers"] as? List<String>) ?: emptyList()
                val blocked = resolveUsers(blockedIds)
                val muted = resolveUsers(mutedIds)
                _uiState.update { it.copy(blockedUsers = blocked, mutedUsers = muted) }
            } catch (_: Exception) {}
        }
    }

    private suspend fun resolveUsers(ids: List<String>): List<BlockedUserInfo> = coroutineScope {
        ids.map { id ->
            async {
                try {
                    val snap = firestore.collection("users").document(id).get().await()
                    val name = snap.getString("name") ?: snap.getString("displayName") ?: "Unknown"
                    val avatarUrl = snap.getString("avatarUrl") ?: snap.getString("avatar")
                    BlockedUserInfo(id, name, avatarUrl)
                } catch (_: Exception) {
                    BlockedUserInfo(id, "Unknown", null)
                }
            }
        }.awaitAll()
    }

    fun update(block: PrivacySettings.() -> PrivacySettings) {
        val updated = _uiState.value.settings.block()
        _uiState.update { it.copy(settings = updated) }
        viewModelScope.launch {
            val uid = auth.currentUser?.uid ?: return@launch
            try {
                firestore.collection("users").document(uid).update("privacySettings", mapOf(
                    "profileVisibility" to updated.profileVisibility,
                    "showActivityOnFeed" to updated.showActivityOnFeed,
                    "showLocationOnMap" to updated.showLocationOnMap,
                    "showStatsToOthers" to updated.showStatsToOthers,
                    "whoCanFollow" to updated.whoCanFollow,
                    "whoCanComment" to updated.whoCanComment,
                    "whoCanSeeClubs" to updated.whoCanSeeClubs,
                )).await()
            } catch (_: Exception) {}
        }
    }

    fun unblockUser(userId: String) {
        val previous = _uiState.value.blockedUsers
        _uiState.update { it.copy(blockedUsers = it.blockedUsers.filter { u -> u.id != userId }) }
        viewModelScope.launch {
            val uid = auth.currentUser?.uid ?: return@launch
            try {
                firestore.collection("users").document(uid)
                    .update("blocked", FieldValue.arrayRemove(userId)).await()
            } catch (_: Exception) {
                _uiState.update { it.copy(blockedUsers = previous) }
            }
        }
    }

    fun unmuteUser(userId: String) {
        val previous = _uiState.value.mutedUsers
        _uiState.update { it.copy(mutedUsers = it.mutedUsers.filter { u -> u.id != userId }) }
        viewModelScope.launch {
            val uid = auth.currentUser?.uid ?: return@launch
            try {
                firestore.collection("users").document(uid)
                    .update("mutedUsers", FieldValue.arrayRemove(userId)).await()
            } catch (_: Exception) {
                _uiState.update { it.copy(mutedUsers = previous) }
            }
        }
    }
}

@Composable
fun PrivacyControlsScreen(
    onBack: () -> Unit = {},
    viewModel: PrivacyViewModel = hiltViewModel(),
) {
    val uiState by viewModel.uiState.collectAsStateWithLifecycle()
    val settings = uiState.settings

    Column(
        modifier = Modifier.fillMaxSize().background(RuvoColors.background).verticalScroll(rememberScrollState()),
    ) {
        Row(
            modifier = Modifier.fillMaxWidth().padding(horizontal = RuvoSpacing.md, vertical = RuvoSpacing.cardGap),
            verticalAlignment = Alignment.CenterVertically,
        ) {
            IconButton(onClick = onBack) { Icon(Icons.Default.ArrowBack, contentDescription = "Back", tint = RuvoColors.textPrimary) }
            Text("Privacy Controls", style = MaterialTheme.typography.headlineSmall, color = RuvoColors.textPrimary, fontWeight = FontWeight.Bold)
        }

        PrivacySection("Profile Visibility") {
            PrivacySelect(
                label = "Who can see my profile",
                desc = "Control who can view your profile and achievements",
                options = listOf("public" to "Public", "friends" to "Friends Only", "private" to "Private"),
                value = settings.profileVisibility,
                onSelect = { viewModel.update { copy(profileVisibility = it) } },
            )
        }

        PrivacySection("Activity Settings") {
            PrivacyToggle(
                label = "Show Activity on Feed",
                desc = "Allow your runs to appear on the community feed",
                value = settings.showActivityOnFeed,
                onToggle = { viewModel.update { copy(showActivityOnFeed = it) } },
            )
            PrivacyToggle(
                label = "Show Location on Map",
                desc = "Display your GPS route on public feeds",
                value = settings.showLocationOnMap,
                onToggle = { viewModel.update { copy(showLocationOnMap = it) } },
            )
            PrivacyToggle(
                label = "Show Stats to Others",
                desc = "Allow others to see your detailed statistics",
                value = settings.showStatsToOthers,
                onToggle = { viewModel.update { copy(showStatsToOthers = it) } },
            )
        }

        PrivacySection("Social Permissions") {
            PrivacySelect(
                label = "Who can follow me",
                desc = "Control who can follow your profile",
                options = listOf("everyone" to "Everyone", "friends" to "Friends Only", "nobody" to "Nobody"),
                value = settings.whoCanFollow,
                onSelect = { viewModel.update { copy(whoCanFollow = it) } },
            )
            PrivacySelect(
                label = "Who can comment",
                desc = "Control who can comment on your runs",
                options = listOf("everyone" to "Everyone", "friends" to "Friends Only", "nobody" to "Nobody"),
                value = settings.whoCanComment,
                onSelect = { viewModel.update { copy(whoCanComment = it) } },
            )
            PrivacySelect(
                label = "Who can see my clubs",
                desc = "Control who can see which clubs you've joined",
                options = listOf("everyone" to "Everyone", "friends" to "Friends Only"),
                value = settings.whoCanSeeClubs,
                onSelect = { viewModel.update { copy(whoCanSeeClubs = it) } },
            )
        }

        PrivacySection("Blocked Users") {
            if (uiState.blockedUsers.isEmpty()) {
                PrivacyEmptyState(icon = Icons.Default.Shield, text = "No blocked users")
            } else {
                uiState.blockedUsers.forEach { user ->
                    BlockedUserRow(
                        user = user,
                        actionLabel = "Unblock",
                        actionColor = RuvoColors.error,
                        actionTextColor = Color.White,
                        onAction = { viewModel.unblockUser(user.id) },
                    )
                }
            }
        }
        Text(
            "Blocked users cannot see your profile, follow you, or interact with your content.",
            style = MaterialTheme.typography.bodySmall,
            color = RuvoColors.textTertiary,
            modifier = Modifier.padding(start = RuvoSpacing.md, end = RuvoSpacing.md, bottom = RuvoSpacing.sm),
        )

        PrivacySection("Muted Users") {
            if (uiState.mutedUsers.isEmpty()) {
                PrivacyEmptyState(icon = Icons.Default.VolumeOff, text = "No muted users")
            } else {
                uiState.mutedUsers.forEach { user ->
                    BlockedUserRow(
                        user = user,
                        actionLabel = "Unmute",
                        actionColor = RuvoColors.surfaceElev,
                        actionTextColor = RuvoColors.textPrimary,
                        onAction = { viewModel.unmuteUser(user.id) },
                    )
                }
            }
        }
        Text(
            "Muted users will not appear on your feed, but they can still see your profile and interact with you.",
            style = MaterialTheme.typography.bodySmall,
            color = RuvoColors.textTertiary,
            modifier = Modifier.padding(horizontal = RuvoSpacing.md),
        )

        Spacer(Modifier.height(80.dp))
    }
}

@Composable
private fun PrivacySection(title: String, content: @Composable ColumnScope.() -> Unit) {
    Column(modifier = Modifier.padding(horizontal = RuvoSpacing.md, vertical = RuvoSpacing.sm)) {
        Text(title.uppercase(), style = MaterialTheme.typography.labelSmall, color = RuvoColors.textTertiary, modifier = Modifier.padding(start = RuvoSpacing.xs, bottom = RuvoSpacing.xs))
        Surface(shape = RoundedCornerShape(RuvoRadius.md), color = RuvoColors.surface, border = BorderStroke(1.dp, RuvoColors.border)) {
            Column { content() }
        }
    }
}

@Composable
private fun PrivacyEmptyState(icon: androidx.compose.ui.graphics.vector.ImageVector, text: String) {
    Column(
        modifier = Modifier.fillMaxWidth().padding(vertical = RuvoSpacing.xl),
        horizontalAlignment = Alignment.CenterHorizontally,
    ) {
        Icon(icon, contentDescription = null, tint = RuvoColors.textTertiary, modifier = Modifier.size(36.dp))
        Spacer(Modifier.height(RuvoSpacing.sm))
        Text(text, style = MaterialTheme.typography.bodySmall, color = RuvoColors.textTertiary)
    }
}

@Composable
private fun BlockedUserRow(
    user: BlockedUserInfo,
    actionLabel: String,
    actionColor: Color,
    actionTextColor: Color,
    onAction: () -> Unit,
) {
    Row(
        modifier = Modifier.fillMaxWidth().padding(horizontal = RuvoSpacing.md, vertical = RuvoSpacing.cardGap),
        verticalAlignment = Alignment.CenterVertically,
        horizontalArrangement = Arrangement.SpaceBetween,
    ) {
        Row(verticalAlignment = Alignment.CenterVertically, horizontalArrangement = Arrangement.spacedBy(RuvoSpacing.cardGap)) {
            if (user.avatarUrl != null) {
                AsyncImage(
                    model = user.avatarUrl,
                    contentDescription = user.name,
                    contentScale = ContentScale.Crop,
                    modifier = Modifier.size(40.dp).clip(CircleShape).background(RuvoColors.surfaceElev),
                )
            } else {
                Box(
                    modifier = Modifier.size(40.dp).clip(CircleShape).background(RuvoColors.surfaceElev),
                    contentAlignment = Alignment.Center,
                ) {
                    Text(
                        user.name.take(1).uppercase(),
                        style = MaterialTheme.typography.bodyMedium,
                        color = RuvoColors.textSecondary,
                        fontWeight = FontWeight.Bold,
                    )
                }
            }
            Text(user.name, style = MaterialTheme.typography.bodyMedium, color = RuvoColors.textPrimary, fontWeight = FontWeight.SemiBold)
        }
        Surface(
            onClick = onAction,
            shape = RoundedCornerShape(RuvoRadius.sm),
            color = actionColor,
        ) {
            Text(
                actionLabel,
                style = MaterialTheme.typography.labelMedium,
                color = actionTextColor,
                fontWeight = FontWeight.SemiBold,
                modifier = Modifier.padding(horizontal = RuvoSpacing.md, vertical = RuvoSpacing.xs),
            )
        }
    }
}

@Composable
private fun PrivacyToggle(label: String, desc: String, value: Boolean, onToggle: (Boolean) -> Unit) {
    Row(
        modifier = Modifier.fillMaxWidth().padding(horizontal = RuvoSpacing.md, vertical = RuvoSpacing.cardGap),
        verticalAlignment = Alignment.CenterVertically,
        horizontalArrangement = Arrangement.spacedBy(RuvoSpacing.cardGap),
    ) {
        Column(modifier = Modifier.weight(1f)) {
            Text(label, style = MaterialTheme.typography.bodyMedium, color = RuvoColors.textPrimary, fontWeight = FontWeight.Medium)
            Text(desc, style = MaterialTheme.typography.bodySmall, color = RuvoColors.textTertiary)
        }
        Switch(
            checked = value,
            onCheckedChange = onToggle,
            colors = SwitchDefaults.colors(
                checkedThumbColor = Color.Black,
                checkedTrackColor = RuvoColors.lime,
                uncheckedThumbColor = RuvoColors.textTertiary,
                uncheckedTrackColor = RuvoColors.surfaceElev,
            ),
        )
    }
}

@Composable
private fun PrivacySelect(
    label: String,
    desc: String,
    options: List<Pair<String, String>>,
    value: String,
    onSelect: (String) -> Unit,
) {
    var expanded by remember { mutableStateOf(false) }
    val displayLabel = options.find { it.first == value }?.second ?: value

    Column(modifier = Modifier.fillMaxWidth().padding(horizontal = RuvoSpacing.md, vertical = RuvoSpacing.cardGap)) {
        Row(
            modifier = Modifier.fillMaxWidth().clickable { expanded = !expanded },
            verticalAlignment = Alignment.CenterVertically,
        ) {
            Column(modifier = Modifier.weight(1f)) {
                Text(label, style = MaterialTheme.typography.bodyMedium, color = RuvoColors.textPrimary, fontWeight = FontWeight.Medium)
                Text(desc, style = MaterialTheme.typography.bodySmall, color = RuvoColors.textTertiary)
            }
            Row(verticalAlignment = Alignment.CenterVertically) {
                Text(displayLabel, style = MaterialTheme.typography.labelMedium, color = RuvoColors.lime, fontWeight = FontWeight.SemiBold)
                Icon(if (expanded) Icons.Default.ExpandLess else Icons.Default.ExpandMore, contentDescription = null, tint = RuvoColors.textTertiary)
            }
        }

        if (expanded) {
            Spacer(Modifier.height(RuvoSpacing.sm))
            options.forEach { (v, l) ->
                Row(
                    modifier = Modifier.fillMaxWidth().clickable { onSelect(v); expanded = false }.padding(vertical = RuvoSpacing.sm, horizontal = RuvoSpacing.xs),
                    verticalAlignment = Alignment.CenterVertically,
                    horizontalArrangement = Arrangement.spacedBy(RuvoSpacing.sm),
                ) {
                    Text(l, style = MaterialTheme.typography.bodySmall, color = RuvoColors.textPrimary, modifier = Modifier.weight(1f))
                    if (v == value) Icon(Icons.Default.Check, contentDescription = null, tint = RuvoColors.lime, modifier = Modifier.size(18.dp))
                }
            }
        }
    }
}
